### ISSUE TITLE
[FEATURE] volume-basic-test.sh should be SKIP if no PGA controls are available

### ISSUE THREAD

**Issue by plbossart** [thesofproject/sof-test#844]
Found with simplified HDaudio passthrough topologies used for IPC4, volume-basic-test fails when there are no PGA components and hence no volume control capabilities.
Well, duh, this should be SKIP not FAIL!

````
 starting test_volume 
+ test_volume
+ /root/sof-test/test-case/volume-basic-test.sh -l 1
2022-01-28 00:24:24 UTC Sub-Test: [INFO] /root/sof-test/test-case/volume-basic-test.sh will use topology /usr/lib/firmware/intel/sof-tplg-v2.0/sof-hda-generic.tplg to run the test case
2022-01-28 00:24:24 UTC Sub-Test: [INFO] Pipeline list to ignore is specified, will ignore 'pcm=HDA Digital' in test case
2022-01-28 00:24:24 UTC Sub-Test: [INFO] Run command to get pipeline parameters
2022-01-28 00:24:24 UTC Sub-Test: [COMMAND] sof-tplgreader.py /usr/lib/firmware/intel/sof-tplg-v2.0/sof-hda-generic.tplg -f 'type:playback' -b ' pcm:HDA Digital' -s 0 -e
2022-01-28 00:24:25 UTC Sub-Test: [INFO] SOF logs collection globally disabled by SOF_LOGGING=none
2022-01-28 00:24:25 UTC Sub-Test: [COMMAND] aplay -D hw:0,0 -c 2 -r 48000 -f S16_LE /dev/zero &
Playing raw data '/dev/zero' : Signed 16 bit Little Endian, Rate 48000 Hz, Stereo
2022-01-28 00:24:26 UTC Sub-Test: [INFO] pgalist number = 0
2022-01-28 00:24:26 UTC Sub-Test: [ERROR] No PGA control is available
2022-01-28 00:24:26 UTC Sub-Test: [ERROR] Starting func_exit_handler(), exit status=1, FUNCNAME stack:
/root/sof-test/case-lib/hijack.sh: line 23: 13211 Killed                  aplay -D $dev -c $channel -r $rate -f $fmt /dev/zero
2022-01-28 00:24:26 UTC Sub-Test: [ERROR]  func_error_exit()  @  /root/sof-test/test-case/volume-basic-test.sh
2022-01-28 00:24:26 UTC Sub-Test: [ERROR]  main()  @  /root/sof-test/test-case/volume-basic-test.sh:62
2022-01-28 00:24:26 UTC Sub-Test: [INFO] Test Result: FAIL!
````

**Comment by marc-hb** [comment]
> Found with simplified HDaudio passthrough topologies used for IPC4

@ranj063 what is the quickest way to reproduce this failure without cherry-picking any work in progress or making complex configuration changes? In other words, what is the quickest way to test a PGA-free topology when starting from everything configured as default?

**Comment by plbossart** [comment]
The instructions are in your email @marc-hb. You will need an HDaudio platform such as TGL laptop. It may work on UpExtreme as well (I haven't tested myself yet)

**Comment by plbossart** [comment]
even we do have gain control, we seem to have different names that don't follow IPC3 conventions, e.g.

````
numid=39,iface=MIXER,name='gain.1.1 Main gain playback 0'
numid=40,iface=MIXER,name='gain.2.1 Main gain capture 0'
numid=41,iface=MIXER,name='gain.3.1 Main gain playback 1'
numid=42,iface=MIXER,name='gain.4.1 Main gain capture 1'
numid=43,iface=MIXER,name='gain.5.1 Main gain playback 2'
numid=44,iface=MIXER,name='gain.6.1 Main gain playback 3'
numid=45,iface=MIXER,name='gain.7.1 Main gain playback 4'
````

That seems like a topology issue and a need for standardizing control names, a recurring theme in audio....

@ranj063 @RanderWang @bardliao @lgirdwood FYI

**Comment by marc-hb** [comment]
Fix submitted in #856 

Thanks @plbossart and especially @ranj063 for holding my shaky IP4 hand.
